### PR TITLE
bookstore

### PR DESCRIPTION
**Pull request by MaVaYuran**

### CODE REVIEW

File: src/main/java/by/mariayuran/BookStore.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+
+    public Order createOrder() {
+        orders = new ArrayList<>();
+        Order newOrder = new Order(orders.size() + 1);
+        createOrderList(newOrder);
+        orders.add(newOrder);
+        return newOrder;
+    }
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/BookStore.java:25]: why inside this merhod we also call 
createOrderList(newOrder);?

File: src/main/java/by/mariayuran/BookStore.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        List<Order> sortedOrders = new ArrayList<>(orders);
+
+        Comparator<Order> comparator = switch (sortBy) {
+            case "id" -> Comparator.comparingInt(Order::getOrderId);
+            case "totalPrice" -> Comparator.comparingDouble(Order::getTotalPrice);
+            case "openingTimestamp" -> Comparator.comparing(Order::getOpeningTimestamp);
+            case "closingTimestamp" -> Comparator.comparing(Order::getClosingTimestamp);
+            default -> Comparator.comparing(order -> order.getStatus().name());
+        };
+        Collections.sort(sortedOrders, comparator);
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/BookStore.java:73]: 👍

File: src/main/java/by/mariayuran/Order.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    public OrderStatus getStatus() {
+        return status;
+    }
+
+    public void setStatus(OrderStatus status) {
+        this.status = status;
+    }
+
+    public void addBook(Book book) {
+        books = new ArrayList<>();
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/Order.java:51]: in each call ypu ypu new List. make books = new ArrayList<>(); onve on the field level

File: src/main/java/by/mariayuran/BookStore.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+    public void completeOrder(int orderId) {
+        Order order = findOrder(orderId);
+        if (order != null)
+            order.makeOrderCompleted();
+    }
+
+    public Book getAnyBook(List<Book> books) {
+        Random random = new Random();
+        int bookId = random.nextInt(books.size() - 1);
```
Comment (Alexmansar) [inline src/main/java/by/mariayuran/BookStore.java:43]: toreBooks.size(); ?

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@MaVaYuran 
please, check comments

File: src/main/java/by/mariayuran/BookStore.java (hunk 4 above)
Comment (MaVaYuran) [reply src/main/java/by/mariayuran/BookStore.java:43]: I dont understand

Comment (Alexmansar) [reply src/main/java/by/mariayuran/BookStore.java:45]: storeBooks.size();

Comment (Alexmansar) [reply src/main/java/by/mariayuran/BookStore.java:45]: You use it as paaramert. but in this vlass you have this field as well. or not?

**Review by Alexmansar — APPROVED** [review]
@MaVaYuran 
Please, merge it

**Pull request merged**